grammar.js

```diff
@@ -72,8 +72,10 @@ module.exports = grammar({
     $._ml4_string_chars,
     $._ml5_string_chars,
     $._ml6_string_chars,
-    $._open_square_bracket,
-    $._open_entry_bracket,
+    // '[' without preceding newline or semicolon
+    $._open_subscript_bracket,
+    // '(' without preceding newline or semicolon
+    $._open_argument_paren,
```

Comment:
Pkl disallows semicolons and newlines before argument lists, and before subscripts. This disambiguates grammar like `foo { bar ; (1) }`, which is meant to be parsed as two elements, and not a method call.

Normally, `;` and `\n` are treated as whitespace. But in these positions, they are important parts of the parse tree.

This introduces external scanner rules that describe the `(` and `[` tokens, which are _not_ preceded by a newline or a semicolon. This matches the existing rules that we have in ANTLR.